### ISSUE TITLE
Use AsciiDoc variables to refer to Minishift generically in documentation

### ISSUE THREAD

**Issue by robin-owen** [minishift/minishift#1547]
Currently, the name of the project is embedded in the AsciiDoc sources by name. To assist in making the documentation project-agnostic, I suggest adding a `project` variable to the Rake-generated **variables.adoc** files.

This would cause changes across a number of files, most notably by requiring `include::variables.adoc[]` in each file of the documentation and replacing all references to "Minishift" as a project with "{project}".

These changes seem reasonable to me -- any thoughts?

**Comment by robin-owen** [comment]
While investigating this, I also noticed that the Rakefile currently in use creates a new **variables.adoc** file for each section of the documentation during the build process. Would it be more efficient to create a directory for common files in **docs/sources** which can be symbolically linked within each section?

Tossed-together example:

```sh
mkdir docs/sources/common_content && cd docs/sources
for d in contributing getting_started openshift troubleshooting using ; do
  pushd $d && ln -s ../common_content common_content && popd
done
```

From here, we could just replace "variables.adoc" with "common_content/variables.adoc" in each file (while noting in **contributing/writing_docs.adoc** the usage of **common_content**).

This would also make for a few changes in the Rakefile so that the build only creates **variables.adoc** in **common_content**.

Let me know what you think.

**Comment by hferentschik** [comment]
> Currently, the name of the project is embedded in the AsciiDoc sources by name. To assist in making the documentation project-agnostic, I suggest adding a project variable to the Rake-generated variables.adoc files.

+1

> This would cause changes across a number of files, most notably by requiring include::variables.adoc[] in each file of the documentation and replacing all references to "Minishift" as a project with "{project}".

Sure. We already discussed once that we can make the project name a variable

**Comment by hferentschik** [comment]
> While investigating this, I also noticed that the Rakefile currently in use creates a new variables.adoc file for each section of the documentation during the build process. 

Right, this is due to some incompatibilities between how AsciiBinder works (there is a bug report/ feature request in the AsciiBinder project) and Middleman. Really there should be only one single file and this should be referenced. However, for some reason in AsciiBinder includes must be relative to the root of the publication. We had a workaround, but had to remove it, in order to be able to have image includes as sub-directories of topics. 

We could try to go deeper into the Ruby code of Middleman and try to hack something, but the easier approach is to just write the file into each topic directory. After all it is generated and the content is defined in a single place (the Makefile).

> Would it be more efficient to create a directory for common files in docs/sources which can be symbolically linked within each section?

What problem do you try to solve? What do you mean with more efficient? As said, the file is generated and defined at a single place. What do you gain with a symbolic link? My main concern is that symbolic link is Unix specific. making it then impossible to build the docs on a Windows machine. 

Right now, I don't see the need to change anything. If anything, I'd rather see a change in AsciiBinder so that one can go to a single include file as it should be.

**Comment by robin-owen** [comment]
> Right, this is due to some incompatibilities between how AsciiBinder works (there is a bug report/ feature request in the AsciiBinder project) and Middleman. Really there should be only one single file and this should be referenced. However, for some reason in AsciiBinder includes must be relative to the root of the publication. We had a workaround, but had to remove it, in order to be able to have image includes as sub-directories of topics.

Ah, I wasn't aware that this was being done to work with AsciiBinder and Middleman -- that makes perfect sense. Given that, I don't see the need to make changes to a build process which is working for everyone.

> What problem do you try to solve?

In my case, having started with this quite recently, it took me a little while to understand what the build was doing with **variables.adoc**, and I had to dive into the Rakefile more deeply than I expected. If that initial friction can be reduced, I see that as a good thing.

My thoughts were along the lines of being able to define static attributes in a version-controlled **common_content/variables.adoc** file and have dynamic attributes generated by the Rakefile in **common_content/generated_variables.adoc** (or similar).

If the concepts of static attributes and dynamic ones were separated, writers would not have to go into the Rakefile in order to change static attributes. Admittedly, attributes should not be added or modified often, but when they are, it's currently required to go into the build process itself rather than modify a clearly-visible AsciiDoc file.

If you think this is worth thinking about more, I can open a dedicated issue for discussion around it.

> What do you mean with more efficient?

By "more efficient," I simply meant creating one file versus six (or more as sections are added over time). This seems cleaner, but I wasn't certain, thus the question about whether it would be more efficient or not. The answer is "no" considering the reasons behind the current way of doing things. :)

> My main concern is that symbolic link is Unix specific. making it then impossible to build the docs on a Windows machine.

Windows has support for them, but it's much more involved than I thought it would be. Given that information, I agree, using regular files is preferable to make the docs build predictable across environments.

> Right now, I don't see the need to change anything. If anything, I'd rather see a change in AsciiBinder so that one can go to a single include file as it should be.

Agreed.

**Comment by hferentschik** [comment]
Merged via pull request #1553
